src/jet/candidate.rs

```diff
@@ -17,32 +25,36 @@ pub struct CandidateResponse {
     bytes_recv: u64,
 }
 
-impl From<Candidate> for CandidateResponse {
-    fn from(c: Candidate) -> Self {
-        let (bytes_sent, bytes_recv) =
-            c.client_transport.map(|transport|(transport.get_nb_bytes_read(), transport.get_nb_bytes_written()))
-                .unwrap_or((0, 0));
+impl From<&Candidate> for CandidateResponse {
+    fn from(c: &Candidate) -> Self {
+        let (bytes_sent, bytes_recv) = match (c.client_nb_bytes_read(), c.client_nb_bytes_written()) {
+                (Some(client_bytes_sent), Some(client_bytes_recv))  => {
+                    (client_bytes_sent, client_bytes_recv)
+                }
+                _ => (0, 0)
+            };
 
         CandidateResponse {
             id: c.id,
-            url: c.url,
-            state: c.state,
-            association_id: c.association_id,
-            transport_type: c.transport_type,
+            url: c.url.clone(),
+            state: c.state.clone(),
+            association_id: c.association_id.clone(),
+            transport_type: c.transport_type.clone(),
             bytes_sent,
             bytes_recv,
         }
     }
 }
-#[derive(Clone)]
+
 pub struct Candidate {
```

Comment:
Because a candidate is a server, and there can be many of them until a client selects a specific server. After the client will do it, the transport is no longer needed (only nb_bytes_read / nb_bytes_written).
Therefore, I left here only the transport field, in which the server transport will be. After the client selects a candidate, the transport will be taken from the candidate, and corresponding copies of the client will be set in client_nb_bytes_read / written.